pkgs/development/python-modules/philipstv/default.nix

```diff
@@ -0,0 +1,67 @@
+{
+  lib,
+  fetchFromGitHub,
+  buildPythonPackage,
+  pythonOlder,
+  poetry-core,
+  poetry-dynamic-versioning,
+  installShellFiles,
+  pytestCheckHook,
+  requests-mock,
+  requests,
+  pydantic,
+  click,
+  appdirs,
+  nix-update-script,
+}:
+
+buildPythonPackage rec {
+  pname = "philipstv";
+  version = "2.1.1";
+  format = "pyproject";
+
+  disabled = pythonOlder "3.9";
+
+  src = fetchFromGitHub {
+    owner = "bcyran";
+    repo = "philipstv";
+    rev = version;
+    hash = "sha256-BvQurZls9NjtHhTXLQ9t8fHkAF/QU/c6mmRvNmE0v90=";
+  };
+
+  nativeBuildInputs = [
+    poetry-core
+    poetry-dynamic-versioning
+    installShellFiles
+  ];
+
+  nativeCheckInputs = [
+    pytestCheckHook
+    requests-mock
+  ];
+
+  propagatedBuildInputs = [
+    requests
+    pydantic
+    click
+    appdirs
+  ];
+
+  postInstall = ''
+    installShellCompletion --cmd philipstv \
+      --bash <(_PHILIPSTV_COMPLETE=bash_source $out/bin/philipstv) \
+      --zsh <(_PHILIPSTV_COMPLETE=zsh_source $out/bin/philipstv) \
+      --fish <(_PHILIPSTV_COMPLETE=fish_source $out/bin/philipstv)
+  '';
+
```

Comment:
Better to add `pythonImportsCheck`

https://nixos.org/manual/nixpkgs/unstable/#using-pythonimportscheck